src/data/fetch.ts

```diff
@@ -0,0 +1,77 @@
+import * as fs from "fs";
+import fetch from "node-fetch";
+
+interface Currency {
+  code: string;
+  name: string;
+  symbol: string;
+}
+interface Language {
+  iso639_1: string;
+  iso639_2: string;
+  name: string;
+  nativeName: string;
+}
+interface RestCountry {
+  name: string;
+  topLevelDomain: string[];
+  alpha2Code: string;
+  alpha3Code: string;
+  callingCodes: string[];
+  capital: string;
+  altSpellings: string[];
+  region: string;
+  subregion: string;
+  population: number;
+  latlng: number[];
+  demonym: string;
+  area: number;
+  gini: number;
+  timezones: string[];
+  borders: string[];
+  nativeName: string;
+  numericCode: string;
+  currencies: Currency[];
+  languages: Language[];
+  translations: {
+    [key: string]: string;
+  };
+  flag: URL["href"];
+  regionalBlocs: [
+    {
```

Comment:
Given that you've added types for `Currency` and `Language` above, it might make sense to also add one for `RegionalBloc` here.